Jenkinsfile

```diff
@@ -51,110 +103,92 @@ def gitMerge(String primaryBranch) {
+    stage('Install') {
       yarnInstall()
     }
 
-    def processes = [:]
-
-    // Browser tests
-    for (int i = 0; i < platforms.size(); i++) {
-      def platform = platforms.get(i)
-
-      def buckets = platform.buckets ?: 1
-      for (int bucket = 1; bucket <= buckets; bucket++) {
-        def suffix = buckets == 1 ? "" : "-" + bucket
-
-        // closure variable - don't inline
-        def c_bucket = bucket
-
-        def name = "${platform.os}-${platform.browser}${suffix}"
-
-        processes[name] = {
-          stage(name) {
-            node("bedrock-${platform.os}") {
-              echo("Bedrock tests for ${name}")
-
-              echo("Checking out code on build node: $NODE_NAME")
-              checkout(scm)
-
-              // windows tends to not have username or email set
-              tinyGit.addAuthorConfig()
-              gitMerge(primaryBranch)
+    stage('Type check') {
+      withEnv(["NODE_OPTIONS=--max-old-space-size=1936"]) {
+        exec("yarn ci-all-seq")
+      }
+    }
 
-              cleanAndInstall()
-              exec("yarn ci")
+    stage('Moxiedoc check') {
+      sh 'yarn tinymce-grunt shell:moxiedoc'
+    }
+  }
+  //
+
+  def platforms = [
+    [ browser: 'chrome', provider: 'aws', buckets: 3 ],
+    [ browser: 'edge', provider: 'aws', buckets: 3 ],
+    [ browser: 'firefox', provider: 'aws', buckets: 3 ],
+    [ browser: 'safari', provider: 'lambdatest', buckets: 2 ]
+  ];
+
+  def processes = [:]
+  def runAllTests = env.BRANCH_NAME == props.primaryBranch
+
+  for (int i = 0; i < platforms.size(); i++) {
+    def platform = platforms.get(i)
+    def buckets = platform.buckets ?: 1
+    for (int bucket = 1; bucket <= buckets; bucket ++) {
+      def suffix = buckets == 1 ? "" : "-" + bucket
+      def s_bucket = "${bucket}"
+      def s_buckets = "${buckets}"
+      def name = "${platform.browser}-${platform.provider}${suffix}"
+      processes[name] = runTestPod(name, platform.browser, platform.provider, s_bucket, s_buckets, runAllTests)
+    }
+  }
 
-              echo("Running browser tests")
-              runBrowserTests(name, platform.browser, platform.os, c_bucket, buckets, runAllTests)
-            }
-          }
+  processes['headless'] = {
+    tinyPods.nodeBrowser([
+      resourceRequestCpu: '6',
+      resourceRequestMemory: '4Gi',
+      resourceLimitCpu: '7.5',
+      resourceLimitMemory: '4Gi'
+    ]) {
+      stage('Test-headless') {
+        yarnInstall()
+        withEnv(["NODE_OPTIONS=--max-old-space-size=1936"]) {
+          sh "yarn ci-all-seq"
         }
       }
-    }
 
-    processes["headless-and-archive"] = {
-      stage("headless tests") {
+      stage('test') {
+        grunt('list-changed-headless')
         runHeadlessTests(runAllTests)
       }
 
-      if (env.BRANCH_NAME != primaryBranch) {
-        stage("Archive Build") {
-          exec("yarn tinymce-grunt prodBuild symlink:js")
+      if (env.BRANCH_NAME != props.primaryBranch) {
```

Comment:
I do have a question here, as I'm not sure how this is supposed to work: we seem to be archiving every other branch instead of *only* the primary branch. Is this by design?

Reply:
Happy to remove the archiving step then?

Reply:
yes, I think so

Reply:
Removed archiving